plugins/CoreHome/vue/src/PeriodSelector/PeriodSelector.vue

```diff
@@ -464,6 +476,44 @@ export default defineComponent({
 
       this.setPiwikPeriodAndDate(this.selectedPeriod, this.dateValue!);
     },
+    updateComparisonValuesFromHash() {
+      this.comparePeriodType = 'previousPeriod';
+      this.compareStartDate = '';
+      this.compareEndDate = '';
+
+      if (!this.isComparing) {
+        return;
+      }
+
+      const comparePeriodType = MatomoUrl.parsed.value.comparePeriodType as string;
+
+      if (!COMPARE_PERIOD_TYPES.includes(comparePeriodType)) {
+        return;
+      }
+
+      const {
+        params: {
+          date: periodDate,
+          period: periodType,
+        },
+      } = ComparisonsStore.getPeriodComparisons().slice(-1).pop() as PeriodComparison;
```

Comment:
This code seems to be selecting the last `PeriodComparison` in an array, but it's not clear from reading why, would code as documentation make it clearer? Maybe with a variable name that indicated what it is/why?

Reply:
The "why" is (was) mostly an "if a user has entered something, only the last value is what he entered".

Most dashboard UI widgets happily accept multiple periods. But as the period selector itself prevents selecting multiple periods (third, fourth, etc. are getting discarded on apply) I switched to a commented `periods[1]`, ignoring user manipulation.